web3_proxy/src/bin/web3_proxy_cli/migrate_stats_to_v2.rs

```diff
@@ -0,0 +1,345 @@
+            rpc_account_shutdown_recevier,
+        )? {
+            // since the database entries are used for accounting, we want to be sure everything is saved before exiting
+            important_background_handles.push(emitter_spawn.background_handle);
+
+            Some(emitter_spawn.stat_sender)
+        } else {
+            None
+        };
+
+        // Basically spawn the full app, look at web3_proxy CLI
+
+        while true {
+            // (1) Load a batch of rows out of the old table until no more rows are left
+            let old_records = rpc_accounting::Entity::find()
+                .filter(rpc_accounting::Column::Migrated.is_null())
+                .limit(number_of_rows_to_process_at_once)
+                .all(db_conn)
+                .await?;
+            if old_records.len() == 0 {
+                // Break out of while loop once all records have successfully been migrated ...
+                warn!("All records seem to have been successfully migrated!");
+                break;
+            }
+
+            // (2) Create request metadata objects to match the old data
+            // Iterate through all old rows, and put them into the above objects.
+            for x in old_records.iter() {
+                // info!("Preparing for migration: {:?}", x);
+
+                // TODO: Split up a single request into multiple requests ...
+                // according to frontend-requests, backend-requests, etc.
+
+                // Get the rpc-key from the rpc_key_id
+                // Get the user-id from the rpc_key_id
+                let authorization_checks = match x.rpc_key_id {
+                    Some(rpc_key_id) => {
+                        let rpc_key_obj = rpc_key::Entity::find()
+                            .filter(rpc_key::Column::Id.eq(rpc_key_id))
+                            .one(db_conn)
+                            .await?
+                            .context("Could not find rpc_key_obj for the given rpc_key_id")?;
+
+                        // TODO: Create authrization
+                        // We can probably also randomly generate this, as we don't care about the user (?)
+                        AuthorizationChecks {
+                            user_id: rpc_key_obj.user_id,
+                            rpc_secret_key: Some(RpcSecretKey::Uuid(rpc_key_obj.secret_key)),
+                            rpc_secret_key_id: Some(
+                                NonZeroU64::new(rpc_key_id)
+                                    .context("Could not use rpc_key_id to create a u64")?,
+                            ),
+                            ..Default::default()
+                        }
+                    }
+                    None => AuthorizationChecks {
+                        ..Default::default()
+                    },
+                };
+
+                // Then overwrite rpc_key_id and user_id (?)
+                let authorization_type = AuthorizationType::Frontend;
```

Comment:
I think `Internal` should be fine doe this

Reply:
What is the difference between the different types? Will change it accordingly